src/script/entity/User.ts

```diff
@@ -113,14 +113,15 @@ export class User {
     };
   }
 
-  constructor(id: string = '') {
+  constructor(id: string = '', domain: string | null) {
```

Comment:
Why do we put explicit `null`s instead of optional `domain` here?

Reply:
We wanted to identify all constructors which potentially need to supply a domain.